tasks/src/main/scala/dagr/tasks/vc/VarDictJava.scala

```diff
@@ -0,0 +1,92 @@
+ * THE SOFTWARE.
+ *
+ */
+
+package dagr.tasks.vc
+
+import java.nio.file.Path
+
+import dagr.commons.CommonsDef.yieldAndThen
+import dagr.commons.io.Io
+import dagr.core.config.Configuration
+import dagr.core.execsystem.{Cores, Memory}
+import dagr.core.tasksystem.{Pipe, PipeOut, ShellCommand, Task}
+import dagr.tasks.DagrDef.{FilePath, PathToBam, PathToFasta, PathToVcf}
+import dagr.tasks.DataTypes.Vcf
+import htsjdk.samtools.SamReaderFactory
+
+/**
+  * The VarDictJava directory structure is assumed to be as follows:
+  *   - Root directory: /path/to/git/VarDictJava
+  *   - Perl scripts:   /path/to/git/VarDictJava/VarDict
+  *   - Startup script: /path/to/git/VarDictJava/build/install/VarDict/bin/VarDict
+  */
+object VarDictJava extends Configuration {
+  /** Config key for the VarDictJava directory.  This should be the root project directory. */
+  val VarDictJavaDirConfigKey = "vardictjava.dir"
+
+  /** Path to the VarDictJava directory. */
+  lazy val VarDictJavaDir = configure[Path](VarDictJavaDirConfigKey)
+
+  /** Path to the VarDictJava install directory with the VarDictJava startup script. */
+  lazy val VarDictJava = VarDictJavaDir.resolve("build/install/VarDict/bin/VarDict")
+
+  /** Path to the VarDictJava bin directory with a multitude of perl scripts. */
+  lazy val VarDictJavaBinDir = VarDictJavaDir.resolve("VarDict")
+}
+
+/**
+  * Tumor-only single-sample variant calling with VarDictJava: https://github.com/AstraZeneca-NGS/VarDictJava
+  */
+class VarDictJava(tumorBam: PathToBam,
+                  bed: FilePath,
+                  ref: PathToFasta,
+                  af: Double = 0.01,
+                  out: PathToVcf = Io.StdOut
+                 ) extends Task with Configuration {
+  override def getTasks: Traversable[_ <: Task] = {
+    val tumorName  = extractSampleName(tumorBam)
+    val vdScript   = VarDictJava.VarDictJava
+    val biasScript = VarDictJava.VarDictJavaBinDir.resolve("teststrandbias.R")
+    val vcfScript  = VarDictJava.VarDictJavaBinDir.resolve("var2vcf_valid.pl")
+
+    val vardict            = new ShellCommand(vdScript.toString, "-G", ref.toString, "-f", af.toString, "-N", tumorName,
+      "-b", tumorBam.toString, "-z", "-c", "1", "-S", "2", "-E", "3", "-g", "4", bed.toString) with PipeOut[Any]
+    val bias               = new ShellCommand(biasScript.toString) with Pipe[Any,Any]
+    val toVcf              = new ShellCommand(vcfScript.toString, "-N", tumorName, "-E", "-f", af.toString) with Pipe[Any,Vcf]
+    val removeRefEqAltRows = new ShellCommand("awk", "{if ($1 ~ /^#/) print; else if ($4 != $5) print}") with Pipe[Vcf,Vcf]
+
+    vardict.requires(Cores(1), Memory("8G"))
+    bias.requires(Cores(0), Memory("32m"))
+    toVcf.requires(Cores(0), Memory("32m"))
+
+    Seq((vardict | bias | toVcf | removeRefEqAltRows > out).withName(this.name))
+  }
+
+  /** Pulls a sample name out of a BAM file. */
+  private def extractSampleName(bam: PathToBam) : String = {
```

Comment:
FYI, the tumor name you supply to vardict can be anything - it's just what it writes to the output file.  So it's fine to extract it like this, or you could just use `tumor` or let the user provide it.